app_dart/lib/src/model/appengine/task.dart

```diff
@@ -129,6 +131,79 @@ class Task extends Model<int> {
     );
   }
 
+  /// Creates a task based on a buildbucket build.
+  static Future<Task> fromBuildbucketResult(
+    bb.Build build,
+    DatastoreService datastore,
+  ) async {
+    log.fine("Creating task from buildbucket result: ${build.toString()}");
+
+    final String repository = build.input!.gitilesCommit!.project!.split('/')[1];
```

Comment:
I would assume that you are trying to split the full repo name here but can you it might make sense to add a comment for others to state specifically what you are doing.